### PR TITLE
Fix for High-DPI bug

### LINKED ISSUE #4555: High DPI causes severe IDE (and app running from IDE) graphics glitches
rd3

Notice the menu bar, oversize combobox dropdown arrow, combobox 'shadow' that appears when its open, and scaling issues in the app (the one running supports high-DPI normally).

If you use the 'Disable DPI scaling' option on the Compatibility tab, things appear normal for that mode, but obviously that's got its own issues; the IDE doesn't support high-DPI and apps running from the IDE would have to implement their own scaling. 

V2.3.0.4221 in VB6 SP6 on Win7x64 150% scaling.

PS- I've encountered this glitch before. If an app running from the IDE uses the TrackPopupMenu[Ex] APIs, it will randomly trigger these distortions in both app and IDE. It's possible Rubberduck as part of its load process calls that or another API that causes the bug. Other similarly complex add-ins I've used do not have this bug however (CodeSMART and MZ-Tools).

### PR DESCRIPTION
**Pull request by mansellan**
Closes #4555.

Note, per-host settings not included in this PR, I think that may be a more wide-ranging change, see discussion on 4555.

### CODE REVIEW

File: Rubberduck.Main/Extension.cs [hunk 1]
```diff
@@ -300,5 +311,15 @@ private void ShutdownAddIn()
                 }
             }
         }
+
+        [DllImport("SHCore.dll", SetLastError = true)]
```
Comment (comintern) [inline Rubberduck.Main/Extension.cs]: Does this mean we won't be able to support Windows 7?

Comment (comintern) [reply Rubberduck.Main/Extension.cs]: I'd also be happier if this were declared elsewhere given that we're planning on rearranging these. Even if it's in one of the other "dll dumps".

File: Rubberduck.Core/UI/Settings/GeneralSettings.xaml [hunk 2]
```diff
@@ -131,6 +131,8 @@
                               IsChecked="{Binding CheckVersionAtStartup}" />
                     <CheckBox Margin="5,0,0,5" Content="{Resx ResxName=Rubberduck.Resources.RubberduckUI, Key=GeneralSettings_CompileBeforeParse}"
                               IsChecked="{Binding CompileBeforeParse}" />
+                    <CheckBox Margin="5,0,0,5" Content="{Resx ResxName=Rubberduck.Resources.RubberduckUI, Key=GeneralSettings_SetDpiUnaware}"
+                              IsChecked="{Binding SetDpiUnaware}" />
```
Comment (bclothier) [inline Rubberduck.Core/UI/Settings/GeneralSettings.xaml]: I'd consider adding a `Enabled` binding to light this up only if it's Windows 8.1 or later (or whatever the DLL we're calling shipped with).

File: Rubberduck.Core/Common/WindowsVersion.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+namespace Rubberduck.Common
+{
+    public struct WindowsVersion : IComparable<WindowsVersion>
+    {
+        public static WindowsVersion Windows10_1809 = new WindowsVersion(10, 0, 17763);
+        public static WindowsVersion Windows10_1803 = new WindowsVersion(10, 0, 17134);
+        public static WindowsVersion Windows10_1709 = new WindowsVersion(10, 0, 16299);
+        public static WindowsVersion Windows10_1703 = new WindowsVersion(10, 0, 15063);
+        public static WindowsVersion Windows10_1607 = new WindowsVersion(10, 0, 14393);
+        public static WindowsVersion Windows10_1511 = new WindowsVersion(10, 0, 10586);
```
Comment (Vogel612) [inline Rubberduck.Core/Common/WindowsVersion.cs]: this implies a code change whenever MS ships a new windows 10. That's untenable. Nobody got time to remember updating this every quarter.

File: Rubberduck.Core/Common/WindowsVersion.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        public static WindowsVersion Windows10 = new WindowsVersion(10, 0, 10240);
+        public static WindowsVersion Windows81_U1 = new WindowsVersion(6, 3, 9600);
+        public static WindowsVersion Windows81 = new WindowsVersion(6, 3, 9200);
+        public static WindowsVersion Windows8 = new WindowsVersion(6, 2, 9200);
+        public static WindowsVersion Windows7_SP1 = new WindowsVersion(6, 1, 7601);
+        public static WindowsVersion Windows7 = new WindowsVersion(6, 1, 7600);
+        public static WindowsVersion WindowsVista_SP2 = new WindowsVersion(6, 0, 6002);
+        public static WindowsVersion WindowsVista_SP1 = new WindowsVersion(6, 0, 6001);
+        public static WindowsVersion WindowsVista = new WindowsVersion(6, 0, 6000);
```
Comment (Vogel612) [inline Rubberduck.Core/Common/WindowsVersion.cs]: I thought we didn't support Vista?

File: Rubberduck.Core/Common/WindowsVersion.cs (hunk 3 above)
Comment (mansellan) [reply Rubberduck.Core/Common/WindowsVersion.cs]: Heh, yeh perhaps I went a little overboard. Back back to major versions only.

File: Rubberduck.Core/Common/WindowsVersion.cs (hunk 4 above)
Comment (mansellan) [reply Rubberduck.Core/Common/WindowsVersion.cs]: Removed any which don't support net46

File: Rubberduck.Main/Extension.cs (hunk 1 above)
Comment (mansellan) [reply Rubberduck.Main/Extension.cs]: Apparantly all that's needed in Windows < 8.1 is to set the global DPI scaling to XP-style (in advanced settings).

File: Rubberduck.Core/UI/Settings/GeneralSettings.xaml (hunk 2 above)
Comment (mansellan) [reply Rubberduck.Core/UI/Settings/GeneralSettings.xaml]: Done.

**Comment by mansellan** [comment]
Hi @fafalone,

There's no API or manifest option available in Windows versions earlier than 8.1, however the glitches can be avoided by selecting Advanced options when setting the system DPI in control panel, then selecting the "Use XP -style scaling" option. Anecdotally, this seems to improve High DPI in all Windows applications, by making text less blurry. I'm not aware of any downsides to doing this, please let me know if you find otherewise.

**Pull request merged**